src/main/java/teammates/ui/webapi/action/PostStudentProfilePictureAction.java

```diff
@@ -0,0 +1,66 @@
+package teammates.ui.webapi.action;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.Part;
+
+import org.apache.http.HttpStatus;
+
+import teammates.common.datatransfer.attributes.StudentProfileAttributes;
+import teammates.common.exception.InvalidHttpRequestBodyException;
+import teammates.common.exception.InvalidParametersException;
+import teammates.common.exception.UnauthorizedAccessException;
+import teammates.common.util.Const;
+import teammates.common.util.GoogleCloudStorageHelper;
+import teammates.ui.webapi.output.StudentProfilePictureResults;
+
+/**
+ * Action: saves the file information of the profile picture that was just uploaded.
+ */
+public class PostStudentProfilePictureAction extends Action {
+    @Override
+    protected AuthType getMinAuthLevel() {
+        return AuthType.LOGGED_IN;
+    }
+
+    @Override
+    public void checkSpecificAccessControl() {
+        if (!userInfo.isStudent) {
+            throw new UnauthorizedAccessException("Student privilege is required to access this resource.");
+        }
+    }
+
+    @Override
+    public ActionResult execute() {
+        try {
+            Part image = req.getPart("studentprofilephoto");
+            if (image == null) {
+                throw new InvalidParametersException(Const.StatusMessages.STUDENT_PROFILE_NO_PICTURE_GIVEN);
```

Comment:
You don't really need to throw `InvalidParametersException` here if this whole `try` block catches it anyway. Just either `throw new InvalidHttpRequestBodyException` or `return new JsonResult(..., HttpStatus.SC_BAD_REQUEST)` directly.